storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -5486,59 +5486,141 @@ static std::string read_file_as_string(const std::string file) {
   fclose(f);
   return std::string(content, len);
 }
-/* Handle DDL for renamed files */
+
+/**
+Handle DDL for renamed files
+example input: test/10.ibd.ren file with content = test/new_name.ibd ;
+      -> tablespace with space_id = 10 will be renamed to test/new_name.ibd
+@return true on success */
 static bool prepare_handle_ren_files(
     const datadir_entry_t &entry, /*!<in: datadir entry */
-    void *arg __attribute__((unused))) {
+    void * /*data*/) {
   if (entry.is_empty_dir) return true;
 
-  std::string ren_file = entry.path;
-  std::string ren_path = entry.rel_path;
-  std::string from_base = entry.datadir;
-  std::string to_base = entry.datadir;
-#ifdef UNIV_DEBUG
-  size_t index = ren_file.find(".ren");
-  assert(index != std::string::npos);
-#endif  // UNIV_DEBUG
-  std::string from = ren_path.substr(0, ren_path.length() - 4);
-  from_base += from;
-  std::string to = read_file_as_string(ren_file);
-  to_base += to;
-  if (to.empty()) {
-    xb::error() << "Can not read " << ren_file;
+  char dest_space_name[FN_REFLEN];
+  space_id_t source_space_id;
+  std::string ren_file_content;
+  std::string ren_file_name = entry.file_name;
+  std::string ren_path = entry.path;
+
+  Fil_path::normalize(ren_path);
+  // trim .ibd.ren
+  source_space_id =
+      atoi(ren_file_name.substr(0, ren_file_name.length() - 8).c_str());
```

Comment:
Please do not use hardcoded number 8 here. Please create the string ".ibd.ren" and use strlen